src/testing/core/mocking/mockLink.ts

```diff
@@ -94,15 +96,37 @@ export class MockLink extends ApolloLink {
 
   public request(operation: Operation): Observable<FetchResult> | null {
     this.operation = operation;
-    const unmatchedVars: Array<Record<string, any> | "<undefined>"> = [];
+    const unmatchedVars: Array<
+      Record<string, any> | "<undefined>" | "<function>"
+    > = [];
     const mocks = this.getMockedResponses(operation);
 
     const index = mocks.findIndex((mock) => {
-      if (mock.variableMatcher(operation.variables)) {
+      const { variables } = mock.request;
+
+      if (typeof variables === "function") {
+        const matched = variables(operation.variables);
+
+        if (!matched) {
+          unmatchedVars.push("<function>");
+        }
+
```

Comment:
```suggestion
    const unmatchedVars: Array<
      Record<string, any> | "<undefined>" | `<function${string}>`
    > = [];
    const mocks = this.getMockedResponses(operation);

    const index = mocks.findIndex((mock) => {
      const { variables } = mock.request;

      if (typeof variables === "function") {
        const matched = variables(operation.variables);

        if (!matched) {
          unmatchedVars.push(
            variables.name ? `<function ${variables.name}>` : "<function>"
          );
        }

```

What do you think about logging a potential function name there?

Reply:
I dig it! Good call